web-ng/src/app/services/data-store/data-store.service.ts

```diff
@@ -73,6 +73,27 @@ export class DataStoreService {
       .set({ title: { en: newTitle } }, { merge: true })
       .then(() => projectId);
   }
+
+  // TODO: Define return types for methods in this class
+  updateProjectLayer(projectId: string, layerId: string, layer: Layer) {
+    const updatedLayer = layer as { id: string };
+    delete updatedLayer['id'];
+    return new Promise((resolve, reject) => {
+      this.db
```

Comment:
`update` already returns a `Promise` - can we remove the `then` `catch` clause and return the Promise directly?